### PR TITLE
feat: allow Houston to wear scarfs

### PR DESCRIPTION
**Pull request by ElianCodes**
Still draft. To be updated following days

Linked to https://github.com/withastro/astro/pull/9476

### CODE REVIEW

File: src/messages/index.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { strip } from '../prompt/util/clear.js';
 
 type Message = string | Promise<string>;
 
-export const say = async (msg: Message | Message[] = [], { clear = false, hat = '', stdin = process.stdin, stdout = process.stdout } = {}) => {
+export const say = async (msg: Message | Message[] = [], { clear = false, clothes = { hat: '', tie: '' }, stdin = process.stdin, stdout = process.stdout } = {}) => {
```
Comment (natemoo-re) [inline src/messages/index.ts]: Even though this is an internal tool, it's best not to break the existing API unless we need to. Keeping `hat` and `tie` separate is a backwards-compatible change.

```suggestion
export const say = async (msg: Message | Message[] = [], { clear = false, hat = '', tie = '', stdin = process.stdin, stdout = process.stdout } = {}) => {
```

File: src/messages/index.ts [hunk 2]
```diff
@@ -48,11 +48,12 @@ export const say = async (msg: Message | Message[] = [], { clear = false, hat =
     const face = (msg: string, { mouth = mouths[0], eye = eyes[0] } = {}) => {
         const [h, v] = walls;
         const [tl, tr, bl, br] = corners;
-        const head = h.repeat(3 - strip(hat).split('').length);
+        const head = h.repeat(3 - strip(clothes.hat).split('').length);
+        const bottom = h.repeat(3 - strip(clothes.tie).split('').length);
```
Comment (natemoo-re) [inline src/messages/index.ts]:
```suggestion
        const head = h.repeat(3 - strip(hat).split('').length);
        const bottom = h.repeat(3 - strip(tie).split('').length);
```

File: src/messages/index.ts [hunk 3]
```diff
@@ -48,11 +48,12 @@ export const say = async (msg: Message | Message[] = [], { clear = false, hat =
...
         return [
-            `${tl}${h.repeat(2)}${hat}${head}${tr}  ${color.bold(color.cyan('Houston:'))}`,
+            `${tl}${h.repeat(2)}${clothes.hat}${head}${tr}  ${color.bold(color.cyan('Houston:'))}`,
```
Comment (natemoo-re) [inline src/messages/index.ts]:
```suggestion
            `${tl}${h.repeat(2)}${hat}${head}${tr}  ${color.bold(color.cyan('Houston:'))}`,
```

File: src/messages/index.ts [hunk 4]
```diff
@@ -48,11 +48,12 @@ export const say = async (msg: Message | Message[] = [], { clear = false, hat =
...
             `${v} ${eye} ${color.cyanBright(mouth)} ${eye}  ${msg}`,
-            `${bl}${h.repeat(5)}${br}`,
+            `${bl}${h.repeat(2)}${clothes.tie}${bottom}${br}`,
```
Comment (natemoo-re) [inline src/messages/index.ts]:
```suggestion
            `${bl}${h.repeat(2)}${tie}${bottom}${br}`,
```

**Comment by ElianCodes** [comment]
Hey @natemoo-re! Thanks for the review! All of your comments should have been addressed!

**Review by ematipico — APPROVED** [review]

**Review by natemoo-re — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**